File: src/Python/66.PlusOne.py
class Solution:
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        n = len(digits)
        i = n - 1
        c = 1
        while i >= 0:
            digit = digits[i]
            digit = digit + c
            digits[i], c = digit % 10, digit // 10
            i -= 1
            if c == 0:
                break
        if c == 1:
            digits.insert(0, c)
        return digits

if __name__ == '__main__':
    solution = Solution()
    print(solution.plusOne([4,3,2,1]))
    print(solution.plusOne([9,9,9,9]))
else:
    pass

File: src/CSharp/1-100/66.PlusOne.cs
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.PlusOne
{

        public class Solution
        {
            public int[] PlusOne(int[] digits)
            {
                int n = digits.Length;
                int i = n - 1;
                int c = 1;
                while (i >= 0)
                {
                    int digit = digits[i];
                    digit += c;
                    digits[i] = digit % 10;
                    c = digit / 10;
                    --i;
                    if (c == 0)
                        break;
                }
                if (c == 1)
                {
                    var list = new List<int>(digits);
                    list.Insert(0, c);
                    return list.ToArray();
                }
                return digits;
            }
        }
}